src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 2 failures in `PowerShell-CI-windows`
Job Cmdlet Tests.Retrieving partial output from jobs.Receive-Job will retrieve partial output, including -Keep results
```powershell
Destination array was not long enough. Check the destination index, length, and the array's lower bounds. (Parameter 'destinationArray')
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Job.Tests.ps1: line 108
108:                         $results = Receive-Job -Keep $job

```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Get-ChildItem -Path -Depth 2 -Name -Force !!!!!!!!
```powershell
Expected 2, but got 1.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 369
369:         1 | Should -Be 2
```